namer/namer.cc

```diff
@@ -1858,10 +1946,17 @@ ast::ParsedFilesOrCancelled defineSymbols(core::GlobalState &gs, vector<SymbolFi
             }
             return ast::ParsedFilesOrCancelled::cancel(move(output), workers);
         }
-        core::MutableContext ctx(gs, core::Symbols::root(), fileFoundDefinitions.tree.file);
-        SymbolDefiner symbolDefiner(move(fileFoundDefinitions.names));
+        auto fref = fileFoundDefinitions.tree.file;
+        core::MutableContext ctx(gs, core::Symbols::root(), fref);
+        auto oldFoundMethodHashes = oldFoundMethodHashesForFiles.find(fref) == oldFoundMethodHashesForFiles.end()
+                                        ? optional<core::FoundMethodHashes>()
+                                        : std::move(oldFoundMethodHashesForFiles[fref]);
+        SymbolDefiner symbolDefiner(move(fileFoundDefinitions.names), move(oldFoundMethodHashes));
```

Comment:
Prefer using the result of `oldFoundMethodHashesForFiles.find(fref)` in the else branch so that we don't have to lookup `fref` twice (which probably requires making this a proper `if` of some sort.